File: Data Structures & Algorithms/valid-sudoku/submission-0.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Map<Integer, HashSet<Character>> rows = new HashMap<>();
        Map<Integer, HashSet<Character>> cols = new HashMap<>();
        Map<String, HashSet<Character>> square = new HashMap<>();
        for (int r = 0; r < 9; r++) {
            for (int c = 0; c < 9; c++) {
                if (board[r][c] == '.') {
                    continue;
                }
                String squareKey = Math.floor(r / 3) + "," + Math.floor(c / 3);
                if (!rows.containsKey(r))  {
                    rows.put(r, new HashSet<>());
                }
                if (!cols.containsKey(c))  {
                    cols.put(c, new HashSet<>());
                }
                if (!square.containsKey(squareKey))    {
                    square.put(squareKey, new HashSet<>());
                }
                if ((rows.get(r)).contains(board[r][c]) || (cols.get(c)).contains(board[r][c])
                        || (square.get(squareKey)).contains(board[r][c]))   {
                    return false;
                }
                rows.get(r).add(board[r][c]);
                cols.get(c).add(board[r][c]);
                square.get(squareKey).add(board[r][c]);
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/surrounded-regions/submission-0.py
class Solution:
    def solve(self, board: List[List[str]]) -> None:
        ROWS = len(board)
        COLS = len(board[0])

        def dfs(r, c):
            if r < 0 or c < 0 or r >= ROWS or c >= COLS or board[r][c] != "O":
                return
            board[r][c] = "T"
            dfs(r + 1, c)
            dfs(r - 1, c)
            dfs(r, c + 1)
            dfs(r, c - 1)

        
        for c in range(COLS):
            for r in range(ROWS):
                dfs(0, c)
                dfs(ROWS - 1, c)
                dfs(r, 0)
                dfs(r, COLS - 1)

        for c in range(COLS):
            for r in range(ROWS):
                if board[r][c] == "O":
                    board[r][c] = "X"

        for c in range(COLS):
            for r in range(ROWS):
                if board[r][c] == "T":
                    board[r][c] = "O"

        